controllers/goharbor/harborcluster/ctrl_setup.go

```diff
@@ -90,3 +94,25 @@ func New(ctx context.Context, name string, configStore *configstore.Store) (comm
 		Log:     ctrl.Log.WithName(application.GetName(ctx)).WithName("controller").WithValues("controller", name),
 	}, nil
 }
+
+// harborClusterPredicateFuncs do filter for events.
+var harborClusterPredicateFuncs = predicate.Funcs{
+	// we do not care other events
+	UpdateFunc: func(event event.UpdateEvent) bool {
+		old, ok := event.ObjectOld.(*goharborv1alpha2.HarborCluster)
+		if !ok {
+			return true
+		}
+
+		new, ok := event.ObjectNew.(*goharborv1alpha2.HarborCluster)
+		if !ok {
+			return true
+		}
+		// when status was not changed and spec was not changed, not need reconcile
+		if equality.Semantic.DeepEqual(old.Spec, new.Spec) && old.Status.Status == new.Status.Status {
```

Comment:
It seems `equality.Semantic.DeepDerivative` is better than ` equality.Semantic.DeepEqua`?

Reply:
updated